dockstore-webservice/src/main/java/io/dockstore/webservice/resources/ToolTesterResource.java

```diff
@@ -0,0 +1,105 @@
+    public ToolTesterResource(DockstoreWebserviceConfiguration configuration) {
+        bucketName = configuration.getToolTesterBucket();
+    }
+
+    @GET
+    @Timed
+    @Path("logs")
+    @Operation(summary = "Get ToolTester log file")
+    @Produces(MediaType.TEXT_PLAIN)
+    public String getToolTesterLog(
+            @QueryParam("tool_id") @Parameter(example = "#workflow/github.com/dockstore/hello_world", required = true) String toolId,
+            @QueryParam("tool_version_name") @Parameter(example = "v1.0.0", required = true) String toolVersionName,
+            @QueryParam("test_filename") @Parameter(example = "hello_world.cwl.json", required = true) String testFilename,
+            @QueryParam("runner") @Parameter(example = "cwltool", required = true) String runner,
+            @QueryParam("log_type") @Parameter(required = true) ToolTesterLogType logType,
+            @QueryParam("filename") @Parameter(example = "1554477737092.log", required = true) String filename) {
+        if (bucketName == null) {
+            throw new CustomWebApplicationException("Dockstore Logging integration is currently not set up",
+                    HttpStatus.SC_SERVICE_UNAVAILABLE);
+        }
+        ToolTesterS3Client toolTesterS3Client = new ToolTesterS3Client(bucketName);
+        try {
+            return toolTesterS3Client.getToolTesterLog(toolId, toolVersionName, testFilename, runner, filename);
+        } catch (AmazonS3Exception e) {
+            throw new CustomWebApplicationException("Dockstore Logging integration is currently not set up",
+                    HttpStatus.SC_SERVICE_UNAVAILABLE);
+        } catch (IOException e) {
+            throw new CustomWebApplicationException("Could not log file contents", HttpStatus.SC_INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    @GET
+    @Timed
+    @Path("logs/search")
+    @Operation(summary = "Search for ToolTester log files")
+    public List<ToolTesterLog> search(
+            @QueryParam("tool_id") @Parameter(example = "#workflow/github.com/dockstore/hello_world", required = true) String toolId,
+            @QueryParam("tool_version_name") @Parameter(example = "v1.0.0", required = true) String toolVersionName) {
+        if (bucketName == null) {
+            throw new CustomWebApplicationException("Dockstore Logging integration is currently not set up",
+                    HttpStatus.SC_SERVICE_UNAVAILABLE);
+        }
+        try {
+            ToolTesterS3Client toolTesterS3Client = new ToolTesterS3Client(bucketName);
+            return toolTesterS3Client.getToolTesterLogs(toolId, toolVersionName);
+        } catch (AmazonS3Exception e) {
+            throw new CustomWebApplicationException("Dockstore Logging integration is currently not set up",
+                    HttpStatus.SC_SERVICE_UNAVAILABLE);
+        } catch (UnsupportedEncodingException e) {
+            throw new CustomWebApplicationException("Could generate s3 bucket key", HttpStatus.SC_INTERNAL_SERVER_ERROR);
```

Comment:
Could **not** generate s3 bucket key?

Also, I would suggest logging both these exceptions with `LOG.error`, and in the previous method too.

You can probably ignore this, but my inclination from previous non-open-source jobs would be to not expose the fact that we are using s3 in the background and return a more generic error message.